airflow/jobs.py

```diff
@@ -535,7 +535,10 @@ def process_dag(self, dag, queue):
             elif ti.is_runnable(flag_upstream_failed=True):
```

Comment:
Yes.. the bug is that the deadlock logic was not waiting for the retry_interval to pass for tasks in the UP_FOR_RETRY state before deciding that a DAG could not be executed and was hence in a deadlocked state. Now, it waits.

The new logic checks if a task that was UP_FOR_RETRY just transitioned to a runnable state, at which point it will execute and the DAG will not be considered deadlocked.